mitodl/mitxonline
# 818 bug unique object id validated contraint on product makes it impossible to add products across different content types

Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/818

#### What's this PR do?
This is to modify the unique key unique_object_id_validated previously added in https://github.com/mitodl/mitxonline/issues/777
to allow the same object id for the course and program

No data issue needs to be fixed on RC as the existing data is still validated with the old and new unique key

#### How should this be manually tested?
The automated test should still pass because it uses purchasable_object, which refers GenericForeignKey("content_type", "object_id"))

To manually test, go to http://mitxonline.odl.local:8013/admin/ecommerce/product/
Add a course and program with the same object id, they both should be allowed now

<img width="873" alt="image" src="https://user-images.githubusercontent.com/3138890/184172478-c4277fd8-d71c-45fa-a213-fce544530890.png">

<img width="878" alt="image" src="https://user-images.githubusercontent.com/3138890/184172627-9d07c4a3-ff7a-44b7-94e9-65fee81998f4.png">

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
(Optional)
